babycare_product_golive_checks/models/product.py

```diff
@@ -0,0 +1,264 @@
+                        reference = []
+                        for l in b.bom_line_ids:
+                            if set([al.id for al in l.attribute_value_ids]).\
+                                    issubset(set([vl.id for vl
+                                             in variant.attribute_value_ids])):
+                                if l.product_qty == 1:
+                                    prodqty = ""
+                                else:
+                                    prodqty0 = int(l.product_qty)
+                                    prodqty = str(prodqty0) + "x "
+                                prodref = l.product_id.default_code
+                                prodappend = prodqty + prodref
+                                reference.append(prodappend)
+                        reference2 = map(str, reference)
+                        reference3 = str(reference2).strip('[]')
+                        reference4 = reference3.replace(',', ' +')
+                        reference5 = reference4.replace("'", "")
+                        if len(reference5) > 64:
+                            error_len_ids.append(template_name)
+                        else:
+                            variant.write({'default_code': reference5})
+                succes_ids.append(template_name)
+            else:
+                error_ids.append(template_name)
+        if succes_ids:
+            text += _("\n\nFollowing config product(s) is/are updated with"
+                      " internal references:\n%s") % (
+                          '\n'.join([i for i in succes_ids]))
+        if error_len_ids:
+            text += _("\n\nFollowing config product(s) is/are not (completely)"
+                      " updated because (some) references are"
+                      " too long:\n%s") % (
+                          '\n'.join([i for i in error_len_ids]))
+        if error_ids:
+            text += _("\n\nFollowing product(s) is/are not updated because"
+                      " they are not config products:\n%s") % (
+                          '\n'.join([i for i in error_ids]))
+        partial = self.env['message.wizard'].create({
+            'text': text
+        })
+        return {
+            'name': _("Message"),
+            'view_mode': 'form',
+            'view_id': False,
+            'view_type': 'form',
+            'res_model': 'message.wizard',
+            'res_id': partial.id,
+            'type': 'ir.actions.act_window',
+            'nodestroy': True,
+            'target': 'new',
+            'domain': '[]',
+        }
+
+    @api.model
+    def update_ean13_images_config_product(self):
+        text = ''
+        succes_ids = []
+        error_ids = []
+        for template_id in self._context.get('active_ids'):
+            obj = self.env['product.template'].browse(template_id)
+            template_name = obj.name
+            boms = obj.bom_ids
+            if obj.type == 'consu' and len(obj.attribute_line_ids) == 1:
```

Comment:
We already have defined a function field bbc_sale/models/product.py. How does this selection relate to that?

Reply:
You're right! I have used this function out of bbc_sale/models/product.py instead.
Thanks.